libraries/concurrent-merkle-tree/src/state.rs

```diff
@@ -0,0 +1,102 @@
+use crate::utils::hash_to_parent;
+
+/// Stores the path of nodes changed in a tree by a Merkle tree operation
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(C)]
+pub struct ChangeLog<const MAX_DEPTH: usize> {
+    /// Historical root value before Path was applied
+    pub root: Node,
+    /// Nodes of off-chain merkle tree
+    pub path: [Node; MAX_DEPTH],
+    /// Bitmap of node parity (used when hashing)
+    pub index: u32,
+    pub _padding: u32,
```

Comment:
Not sure if this makes any difference, but `Path` and `ChangeLog` are exactly the same struct. Is that a coincidence or is something much more beautiful at work?

Reply:
They are 100% equivalent. The duplication is so the code is less confusing semantically. `path != proof`